adding kind install instructions for KIC (#5365)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/deployment/kind.md

```

### Set your Kong Manager password

The Kong Manager UI requires authentication. {{site.kic_product_name}} uses the `kong-enterprise-superuser-password`
secret to set the default value for the default `kong_admin` user.

Run the following, replacing `cloudnative` with a random password of your choice and note it down.

```bash
$ kubectl create secret generic kong-enterprise-superuser-password  -n kong --from-literal=password=cloudnative
```

Once these resources have been created, we are ready to deploy {{site.kic_product_name}}.

## Install Kong

```bash
kubectl apply -f https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/v{{ page.kong_version | replace: ".x", ".0" }}/deploy/single/all-in-one-postgres-enterprise.yaml
```

This may take a few minutes if this is the first time you're running {{ site.base_gateway }} in Docker with a database.

Once bootstrapped, you should see the {{site.kic_product_name}} running:

```bash
$ kubectl get pods -n kong
NAME                            READY   STATUS      RESTARTS   AGE
ingress-kong-548b9cff98-n44zj   2/2     Running     0          21s
kong-migrations-pzrzz           0/1     Completed   0          4m3s

> postgres-0                      1/1     Running     0          4m3s

Reviewer: Please split the result out into a separate codeblock.

Let's create some `ingress` configurations to enable this:

```bash
echo "
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-proxy
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: kong.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-proxy
            port: 
              number: 80
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-admin
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: admin-api.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-admin
            port: 
              number: 80
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-manager
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: manager.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-manager
            port: 
              number: 80
" | kubectl apply -f -
```

> Finally, apply the following patch to set the `KONG_ADMIN_API_URI` to the hostname we set in the ingress above:

Reviewer: ```suggestion
Apply the following patch to set the `KONG_ADMIN_API_URI` to the hostname you set in the ingress above:
```

```bash
kubectl patch deployment -n kong ingress-kong -p "{\"spec\": { \"template\" : { \"spec\" : {\"containers\":[{\"name\":\"proxy\",\"env\": [{ \"name\" : \"KONG_ADMIN_API_URI\", \"value\": \"http://admin-api.127-0-0-1.nip.io\" }]}]}}}}"
```

### Expose the proxy to your host machine

We need to apply some `kind` specific patches to make our proxy accessible on our host machine:

```bash
kubectl patch deployment -n kong ingress-kong -p '{"spec":{"template":{"spec":{"containers":[{"name":"proxy","ports":[{"containerPort":8000,"hostPort":80,"name":"proxy","protocol":"TCP"},{"containerPort":8443,"hostPort":443,"name":"proxy-ssl","protocol":"TCP"}]}]}}}}'
```

It will take a few minutes to roll out the updated deployment. Once the new

> `ingress-kong` pod is up and running, visit <http://manager.127-0-0-1.nip.io/> and you should be able to log

Reviewer: ```suggestion
`ingress-kong` pod is up and running, visit `http://manager.127-0-0-1.nip.io` and you should be able to log
```

  please change this value.

### Kong Enterprise bootstrap password

Next, we need to create a secret containing the password using which we can login into Kong Manager.
Please replace `cloudnative` with a random password of your choice and note it down.

```bash
$ kubectl create secret generic kong-enterprise-superuser-password  -n kong --from-literal=password=cloudnative
secret/kong-enterprise-superuser-password created
```

Once these are created, we are ready to deploy Kong Enterprise
Ingress Controller.

## Install

```bash
kubectl apply -f https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/v{{ page.kong_version | replace: ".x", ".0" }}/deploy/single/all-in-one-postgres-enterprise.yaml
```

It takes a little while to bootstrap the database.
Once bootstrapped, you should see the {{site.kic_product_name}} running with
Kong Enterprise as its core:

```bash
$ kubectl get pods -n kong
NAME                            READY   STATUS      RESTARTS   AGE
ingress-kong-548b9cff98-n44zj   2/2     Running     0          21s
kong-migrations-pzrzz           0/1     Completed   0          4m3s
postgres-0                      1/1     Running     0          4m3s
```

> Apply kind specific patches to forward the hostPorts to the ingress controller, set taint tolerations, and schedule it to the custom labeled node.

Author: Yes, this is requirement for installing on Kind.

> Apply kind specific patches to forward the hostPorts to the ingress controller, set taint tolerations, and schedule it to the custom labeled node.

Author: yes, thats right.

Set up the ingress rules for Admin, Manager and Proxy

```bash
echo "
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-proxy
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: kong.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-proxy
            port: 
              number: 80
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-admin
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: admin-api.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-admin
            port: 
              number: 80
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kong-manager
  namespace: kong
spec:
  ingressClassName: kong
  rules:
  - host: manager.127-0-0-1.nip.io
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: kong-manager
            port: 
              number: 80
" | kubectl apply -f -
```

> Next, if you browse to the IP address or host of the kong-manager service in your Browser, which in our case is <http://manager.127-0-0-1.nip.io/>. Kong Manager should load in your browser. Try logging in to the Manager with the username kong_admin and the password you supplied in the prerequisite, it should fail. The reason being we've not yet told Kong Manager where it can find the Admin API.

Author: Yes thats right. Thats the user name of the admin user.
